### PR TITLE
TRUNK-5387 Allowed override of i18n messages during installation

### PR DESCRIPTION
**Pull request by ajit-junghare**
## Description of what I changed

1. modified class & tests
2. refactored as per comments from Wolf
3. added Test properties files
4. added placeholder properties file with example
5. added license for Test Class

## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5387

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: web/src/test/java/org/openmrs/web/filter/util/CustomResourceLoaderTest.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import org.junit.Assert;
+import org.junit.Test;
+import org.openmrs.util.LocaleUtility;
+
+import java.util.*;
+
+public class CustomResourceLoaderTest {
+
+	@Test
+	public void testUpdateResource() {
```
Comment (gitcliff) [inline web/src/test/java/org/openmrs/web/filter/util/CustomResourceLoaderTest.java:21]: @ajit-junghare the tests method name should be something like updateResources_shouldUpdateResources()

kindly have a look at this https://wiki.openmrs.org/display/docs/Unit+Testing+With+at-should+Annotation

File: web/src/test/java/org/openmrs/web/filter/util/CustomResourceLoaderTest.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    	Assert.assertTrue(availablelocales.contains(expectedLocale)); // "fr" as a language is present
+		//check property value change from XYZ to ABC
+		Assert.assertEquals("ABC",resources.get(expectedLocale).getString("test.install.header.caption"));
+		//check non-override property is unaffected  
+		Assert.assertEquals("PQR",resources.get(expectedLocale).getString("test.install.header2"));
+    }
+	
+
+    @Test
+	public void testUpdateResourceEmptyOverrideProperties() {
```
Comment (gitcliff) [inline web/src/test/java/org/openmrs/web/filter/util/CustomResourceLoaderTest.java:39]: the same here

Comment (ajit-junghare) [reply web/src/test/java/org/openmrs/web/filter/util/CustomResourceLoaderTest.java]: done

File: web/src/test/java/org/openmrs/web/filter/util/CustomResourceLoaderTest.java (hunk 1 above)
Comment (ajit-junghare) [reply web/src/test/java/org/openmrs/web/filter/util/CustomResourceLoaderTest.java]: ok .. done

**Comment by gitcliff** [comment]
@ajit-junghare  could squash your commits as advised here https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Comment by ajit-junghare** [comment]
@gitcliff Could you please comment on what else is needed.

**Comment by gitcliff** [comment]
@ajit-junghare  i see your having 4 commits please squash them

**Comment by ajit-junghare** [comment]
> 

Done

**Pull request closed without merging**